docs/guide/interactions/views.rst

```diff
@@ -0,0 +1,467 @@
+
+.. code-block:: python
+    :emphasize-lines: 7
+
+    class Confirm(discord.ui.View):
+
+        def __init__(self, user: discord.User, *, timeout: Optional[int] = None) -> None:
+            super().__init__(timeout=timeout)
+            self.user: discord.User = user
+            self.result: Optional[bool] = None
+            self.add_item(YesButton())
+
+        async def interaction_check(self, interaction: discord.Interaction) -> bool:
+            return interaction.user == self.user 
+
+
+Decorator-based Components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In reality, our components are relatively simple, so we can use a helper function decorator instead.
+
+In this case, we're creating a button, so we can use the :func:`~discord.ui.button` decorator inside our ``View`` like so:
+
+.. code-block:: python
+    :emphasize-lines: 12-19
+
+    class Confirm(discord.ui.View):
+
+        def __init__(self, user: discord.User, *, timeout: Optional[int] = None) -> None:
+            super().__init__(timeout=timeout)
+            self.user: discord.User = user
+            self.result: Optional[bool] = None
+            self.add_item(YesButton())
+
+        async def interaction_check(self, interaction: discord.Interaction) -> bool:
+            return interaction.user == self.user 
+
+        @discord.ui.button(label='No', style=discord.ButtonStyle.primary)
+        async def no_button(
+            self,
+            interaction: discord.Interaction,
+            button: discord.ui.Button[Confirm],
+        ) -> None:
+            self.result = False
+            self.stop()
+            await interaction.response.defer()
+
+Using the decorator approach greatly simplifies the code, but it's not as flexible as using a custom class.
+We set parameters prior to the creation of the view instance, so context-specific variables are not available.
+If modifications are needed, we would instead have to override component instance attributes in the ``__init__`` method.
+
+The function we're decorating acts similarly to the :meth:`~discord.ui.Button.callback` method, it's called when the button is clicked.
+However, the arguments passed to the function are different. Rather than ``self`` referring to the component, it refers to the view, with
+the component being passed as the last argument.
+
+When using component decorators, we no longer need to explicitly add the component to the view, this is done automatically.
+
+
+Sending views
+--------------
+
+We send views using the ``view`` parameter in methods which send messages, for example :meth:`TextChannel.send`.
+
+In our case we're creating a confirmation prompt for our ban command, so we'll want to use the :meth:`InteractionResponse.send_message` method.
+
+Our command might look like this:
+
+.. code-block:: python
+
+    @discord.app_commands.command()
+    @discord.app_commands.describe(member='The member to ban.')
+    async def ban(interaction: discord.Interaction, member: discord.Member) -> None:
```

Comment:
I personally don't think a ban command is the best example of showing a confirmation view. You can find my concerns in `?tag nabc`. Instead, I recommend making a more simpler command, such as whether does the user want a cookie or not

Reply:
The idea here was to use somewhat practical examples which don't require much additional context. 
I'm not sure I 100% agree with your take on ban commands, (banning external users for example isn't particualrly easy with the UI, (that being said this command doesn't help with that either)).

Reply:
If you think about it, when most people don't understand what a view is, they want an core version of what views do. Like adding a button and getting a response back from it. You could rewrite the section to build a simple view that just counts how much time a button has been pressed, or an button that just responses with "Hello World!" back. It doesn't need to be as complex as a confirmation view tied with a ban command. I'd also recommend adding a note that notes to check out the examples for more detailed information.

Reply:
Reworked this to be a /report command instead